app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -1,102 +1,213 @@
+                            .putInt(getString(R.string.main_page_selected_service), service).apply();
+                    defaultPreferences.edit()
+                            .putString(getString(R.string.main_page_selected_channel_url), url).apply();
+                    defaultPreferences.edit()
+                            .putString(getString(R.string.main_page_selected_channel_name), name).apply();
+
+                    mainPageContentPref.setSummary(name);
                 });
+                selectChannelFragment.setOnCancelListener(() -> {
+                    mainPageContentPref.setSummary(mainPrefOldSummary);
+                    mainPageContentPref.setValue(mainPrefOldValue);
+                });
+                selectChannelFragment.show(getFragmentManager(), "select_channel");
+            } else {
+                mainPageContentPref.setSummary(getMainPageSummeryByKey(newValue));
+            }
+
+            defaultPreferences.edit().putBoolean(Constants.KEY_MAIN_PAGE_CHANGE, true).apply();
+
+            return true;
+        });
+
+        Preference importDataPreference = findPreference(getString(R.string.import_data));
+        importDataPreference.setOnPreferenceClickListener((Preference p) -> {
+            Intent i = new Intent(getActivity(), FilePickerActivityHelper.class)
+                    .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_MULTIPLE, false)
+                    .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_CREATE_DIR, false)
+                    .putExtra(FilePickerActivityHelper.EXTRA_MODE, FilePickerActivityHelper.MODE_FILE);
+            startActivityForResult(i, REQUEST_IMPORT_PATH);
+            return true;
+        });
+
+        Preference exportDataPreference = findPreference(getString(R.string.export_data));
+        exportDataPreference.setOnPreferenceClickListener((Preference p) -> {
+            Intent i = new Intent(getActivity(), FilePickerActivityHelper.class)
+                    .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_MULTIPLE, false)
+                    .putExtra(FilePickerActivityHelper.EXTRA_ALLOW_CREATE_DIR, true)
+                    .putExtra(FilePickerActivityHelper.EXTRA_MODE, FilePickerActivityHelper.MODE_DIR);
+            startActivityForResult(i, REQUEST_EXPORT_PATH);
+            return true;
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (DEBUG) {
+            Log.d(TAG, "onActivityResult() called with: requestCode = [" + requestCode + "], resultCode = [" + resultCode + "], data = [" + data + "]");
+        }
+
+        if ((requestCode == REQUEST_IMPORT_PATH || requestCode == REQUEST_EXPORT_PATH)
+                && resultCode == Activity.RESULT_OK) {
+            String path = data.getData().getPath();
+            if(requestCode == REQUEST_EXPORT_PATH) {
+                exportDatabase(path + "/NewPipeData.zip");
```

Comment:
Maybe we should ask the user whether he wants us to replace an older version of his db.

Reply:
I thught about buttint a timestampt in the filename.

Reply:
I thught about buttint a timestampt in the filename.

Reply:
Sounds good.